### ISSUE TITLE
Reloading html page with unchanged CSS generates traceback

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11143]
|<img alt="jacob's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| jacob reported|
|-|-|
|Trac ID|trac#1619|
|Type|defect|
|Created|2006-04-02 19:42:50Z|

Here is an example, based on the web2 demo file.
To repeat:

  1. Put this file in a directory
  2. Create a CSS file called "schedcss" in the same directory. Suitable content is "h1 {font-size: 10pt}"
  3. Start server with "twistd -ny <filename>
  4. Point browser to "http://localhost:8080/". This should load fine.
  5. Press reload button. This should generate traceback.
  6. Do "touch schedcss".
  7. Press reloadbutton again. This should not generate a traceback.

----------------------------------------------------------------------------------
```python
#!/usr/bin/env python2.4
# Copyright (c) 2001-2004 Twisted Matrix Laboratories.
# See LICENSE for details.

"""I am a simple test resource.
"""
import os.path
import cgi as pycgi

from twisted.web2 import log, iweb
from twisted.web2 import static, wsgi, resource, responsecode, twcgi
from twisted.web2 import resource, stream, http, http_headers
from twisted.internet import reactor

class CSS(resource.Resource):
    def render(self, req):
        # This simulates what happens if the file was unchanged
        # It generates the traceback
        return http.Response(
            304,
            {'content-type': http_headers.MimeType('text', 'css')},
            None)
        # This has a valid stream. We don't get a traceback
        return http.Response(
            304,
            {'content-type': http_headers.MimeType('text', 'css')},
            '')
        # This shows what would be returned if the file is changed 
        return http.Response(
            responsecode.OK,
            {'content-type': http_headers.MimeType('text', 'css')},
            'h1 {font-size: 10pt}')

class Toplevel(resource.Resource):
    addSlash=True

    def render(self, req):
        contents = """<html>
<head><title>Twisted.web2 demo server</title>
<link rel="stylesheet" type="text/css" href="mycss"/>
</head>
<body>

<h1>Hello!</h1>  This is a twisted.web2 demo.
</body>
</html>"""

        return http.Response(
            responsecode.OK,
            {'content-type': http_headers.MimeType('text', 'html')},
            contents)

    
    child_mycss = static.File('./schedcss')
    #child_mycss = CSS()

if __name__ == '__builtin__':
    from twisted.application import service, strports
    from twisted.web2 import server, vhost, channel

    test = Toplevel()

    # Setup default common access logging
    res = log.LogWrapperResource(test)
    log.DefaultCommonAccessLoggingObserver().start()

    # Create the site and application objects
    site = server.Site(res)
    application = service.Application("demo")
    
    # Serve it via standard HTTP on port 8080
    s = strports.service('tcp:8080', channel.HTTPFactory(site))
    s.setServiceParent(application)
```
----------------------------------------------------------------------------------
```
Traceback:
2006/04/02 20:35 CEST [-] twistd 2.2.0 (/usr/bin/python2.3 2.3.5) starting up
...
2006/04/02 20:40 CEST [HTTPChannel,0,127.0.0.1] Traceback (most recent call last):
          File "/usr/lib/python2.3/site-packages/twisted/internet/defer.py", line 307, in _runCallbacks
            self.result = callback(self.result, *args, **kw)
          File "/usr/lib/python2.3/site-packages/twisted/web2/server.py", line 373, in _cbFinishRender
            d.callback(response)
          File "/usr/lib/python2.3/site-packages/twisted/internet/defer.py", line 229, in callback
            self._startRunCallbacks(result)
          File "/usr/lib/python2.3/site-packages/twisted/internet/defer.py", line 294, in _startRunCallbacks
            self._runCallbacks()
        --- <exception caught here> ---
          File "/usr/lib/python2.3/site-packages/twisted/internet/defer.py", line 307, in _runCallbacks
            self.result = callback(self.result, *args, **kw)
          File "/usr/lib/python2.3/site-packages/twisted/web2/http.py", line 415, in writeResponse
            response.stream.close()
          File "/usr/lib/python2.3/site-packages/twisted/web2/log.py", line 42, in close
            self.stream.close()
        exceptions.AttributeError: 'NoneType' object has no attribute 'close'
```

<details><summary>Searchable metadata</summary>

```
trac-id__1619 1619
type__defect defect
reporter__jacob jacob
priority__normal normal
milestone__Web2_Gold_Master Web2-Gold-Master
branch__ 
branch_author__ 
status__closed closed
resolution__duplicate duplicate
component__web2 web2
keywords__ 
time__1144006970000000 1144006970000000
changetime__1167703111000000 1167703111000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)<a name="note_4"></a>|@jyknight set status to `closed`|
|-|-|

Duplicate of the (now fixed) bug [#10999](https://github.com/twisted/twisted/issues/10999).

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
